lib/xtool/xerrhead.cpp

```diff
@@ -47,28 +47,29 @@ void XErrorHandler::RTC(const char *file, unsigned int line, const char *expr)
 
 void XErrorHandler::Abort(const char* message, int code, int val, const char* subj)
 {
+	std::ostringstream stream;
 	time_t now;
     time(&now);
     char time_buf[sizeof "2011-10-08T07:07:09Z"];
     strftime(time_buf, sizeof time_buf, "%Y-%m-%dT%H:%M:%SZ", gmtime(&now));
 
-	log_file.open(log_name.c_str(),std::ios::out|std::ios::app);
-	log_file<<time_buf<<" ";
-	log_file<<"Abort: "<<message<<" code:"<<code<<" val:"<<val<<std::endl;
+	stream<<time_buf<<" "<<std::endl;
+		stream<<GIT_BRANCH<<GITHUB_BRANCH<<" "<<GIT_COMMIT_HASH<<GITHUB_COMMIT_HASH<<std::endl<<std::endl;
+
+stream<<"Abort: "<<message<<" code:"<<code<<" val:"<<val<<std::endl;

```

Comment:
Something wrong with the indent here and in other parts of the changes.

Reply:
fixed